pyomo/core/plugins/transform/strip_bounds.py

```diff
@@ -0,0 +1,68 @@
+"""Transformation to strip variable bounds from a model."""
+import textwrap
+
+from pyomo.core.base.var import Var
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+from pyomo.core.kernel.component_map import ComponentMap
+from six import iteritems
+from pyomo.core.kernel.set_types import Reals
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class VariableBoundStripper(IsomorphicTransformation):
+    """Strips bounds from variables."""
+
+    alias('core.strip_var_bounds',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(VariableBoundStripper, self).__init__()
+
+    def _apply_to(self, instance, strip_domains=True, tmp=False):
+        """Apply the transformation.
+
+        Args:
+            instance (Block): the block on which to strip variable bounds
+            strip_domains (bool, optional): strip the domain for discrete
+                variables as well
+            tmp (bool, optional): Whether the bound stripping will be
+                temporary. If so, store information for reversion.
+
+        Returns:
+            None
+
+        """
+        if tmp and not hasattr(instance, '_tmp_var_bound_strip_lb'):
```

Comment:
I'm not sure I agree with this logic. If `instance` already has one of these component maps then I would expect the transformation to throw an error saying that the transformation has already been applied and must be reverted before it can be applied again. I can't think of a case where you would want to strip the bounds multiple times before reverting unless you are adding variables to the block after the transformation. I would be worried about over-writing information by reusing an existing component map.

Reply:
I agree with Bethany. Qi, any comment?